fe/fe-core/src/main/java/org/apache/doris/nereids/glue/translator/PhysicalPlanTranslator.java

```diff
@@ -1827,9 +1827,15 @@ public PlanFragment visitPhysicalNestedLoopJoin(
 
     @Override
     public PlanFragment visitPhysicalLimit(PhysicalLimit<? extends Plan> physicalLimit, PlanTranslatorContext context) {
+        Plan childPlan = physicalLimit.child(0);
         PlanFragment inputFragment = physicalLimit.child(0).accept(this, context);
         PlanNode child = inputFragment.getPlanRoot();
-        child.setLimit(MergeLimits.mergeLimit(physicalLimit.getLimit(), physicalLimit.getOffset(), child.getLimit()));
+        if (childPlan instanceof PhysicalLimit) {
```

Comment:
the better way fix this problem
1. remove post process rule: org.apache.doris.nereids.processor.post.AddOffsetIntoDistribute
2. translate global limit into exchange, if its child is not a exchange node. refer to org.apache.doris.nereids.glue.translator.PhysicalPlanTranslator#visitPhysicalTopN

Reply:
thx, done